docs/proposals/sharding-by-partitions.md

```diff
@@ -0,0 +1,81 @@
+
+Each of the 30 ingesters would now have 100K series assigned to them. The series in each of the ingesters would be very distinct from each other. Now, if we randomly pick two ingesters out of the 30, there will always be at-least one time series which will not have 2 ingester replicas available. This is the reason behind the remote write errors.
+
+## Solution
+
+The fundamental problem behind unavailability is the sharding scheme used here. Each series is hashed an assigned to a different ingesters resulting in 1 million shards. If we can decrease the total number of shards, this problem can be solved.
+Let us consider the same scenario of assigning 1 million active series to 30 ingesters. Instead of assigning each time series directly to ingesters, the 1 million series will be split into 10 partitions of size 100K. Each partition can then be assigned to three ingesters.
+
+![Solution](/images/proposals/sharding-by-partitions-solution.png)
+
+Now the condition for availability will change to every partition should have at-least two replicas available. The likelihood of the two random ingesters of the tenant being down having the same partitions on them is much less than before.
+
+
+### Probability of unavailability when two ingesters are down
+
+If two ingesters are down in the cluster, what is probability of two replicas of any partition being unavailable can be expressed by the following formulae. This formulae was derived by Mr. Johney George a retired mathematics professor.
+
+![Formulae](/images/proposals/sharding-by-partitions-formulae.png)
+
+n - Number of nodes
+p - Number of partitions
+
+The following graph plots how the probability of unavailability changes with number of partitions in a cluster with 1000 nodes.
+
+![Graph](/images/proposals/sharding-by-partitions-graph.png)
+
+Sharding series by all labels is a special case of this, where number of partitions is 1 million. Beyond 10^5 partitions, there is 100% probability of unavailability if two ingesters are down.
+
+### Splitting series into partitions
+
+A new sharding mechanism is required to calculate the number of partitions and split series to partitions.
+
+To calculate the number of partitions, we can use the following:
+- `num_partitions = CEIL(max_active_series_per_user / partitions_size)`
+
+To split series into partitions there are two options::
+
+* Modulo hashing - `hash (labels) % num_partitions`
+* Rendezvous hashing - See: https://en.wikipedia.org/wiki/Rendezvous_hashing
+
+The recommendation is to use Rendezvous hashing because when the number of partitions changes, the required resharding is minimal similar to consistent hashing. The implementation of rendezvous hashing is much simpler and doesn’t need a ring to operate.
+
+### Assign partitions to ingester
+
+Partitions can be assigned to 3 distinct ingesters using the ingester ring. This will work similar to how three replicas are picked for a given time series.
```

Comment:
Hummm...

I thought the partition would be assigned to its own subring, no?

If i have 10 partition i can only have max 30 ingesters (10*3)? I thought the idea was to divide the tenant subring into #numberOfPartitions subrings:

EX: if the tenantA has a subring of 100 ingesters and 10 partitions, each partition will have 10 ingesters.